examples/with-react-intl/pages/_app.tsx

```diff
@@ -27,7 +27,7 @@ const getInitialProps: typeof App.getInitialProps = async appContext => {
   const {
     ctx: {req},
   } = appContext;
-  const locale = (req as any)?.locale ?? 'en';
+  const locale = (req as any)?.locale || (window as any).LOCALE || 'en';
```

Comment:
Could we use `__NEXT_DATA__` instead?

```suggestion
  const locale = (req as any)?.locale || (window as any).__NEXT_DATA__.props.locale || 'en';
```

Reply:
That looks private so I'm not sure

Reply:
The usage of undocumented internals is highly discouraged, `window` is better here.